### PR TITLE
logs-show: Stop using source monotonic/realtime timestamp in display

### PR DESCRIPTION
**Pull request by daandemeyer**
Not using the same timestamp source for all logs being displayed can result in hard to parse logs because the timestamps of subsequent logs might not be in increasing order. Let's make sure we use the same timestamp source for all the logs we display.

### CODE REVIEW

File: src/shared/logs-show.c [hunk 1]
```diff
@@ -1266,55 +1266,22 @@ static int output_cat(
...
-        }
-        if (r < 0)
-                return r;
-
-        if (realtime)
-                realtime_good = safe_atou64(realtime, &ret_ts->realtime) >= 0;
-        if (!realtime_good || !VALID_REALTIME(ret_ts->realtime))
-                realtime_good = sd_journal_get_realtime_usec(j, &ret_ts->realtime) >= 0;
+        realtime_good = sd_journal_get_realtime_usec(j, &ret_ts->realtime) >= 0;
         if (!realtime_good)
```
Comment (yuwata) [inline src/shared/logs-show.c:1276]: You can drop the boolean variable:
```c
if (sd_journal_get_realtime_usec(...) < 0)
        ret_ts->realtime = USEC_INFINITY;
```

File: src/shared/logs-show.c [hunk 2]
```diff
@@ -1266,55 +1266,22 @@ static int output_cat(
...
-        if (!monotonic_good || !VALID_MONOTONIC(ret_ts->monotonic))
-                monotonic_good = boot_id_good = sd_journal_get_monotonic_usec(j, &ret_ts->monotonic, ret_boot_id) >= 0;
+        monotonic_good = boot_id_good = sd_journal_get_monotonic_usec(j, &ret_ts->monotonic, ret_boot_id) >= 0;
         if (!monotonic_good)
                 ret_ts->monotonic = USEC_INFINITY;
-
-        if (!boot_id_good)
-                boot_id_good = sd_journal_get_monotonic_usec(j, NULL, ret_boot_id) >= 0;
         if (!boot_id_good)
                 *ret_boot_id = SD_ID128_NULL;
```
Comment (yuwata) [inline src/shared/logs-show.c:1283]: Also here.
```c
if (sd_journal_get_monotonic_usec(...) < 0) {
        ...
}
```

**Review by yuwata — CHANGES_REQUESTED** [review]
The initial half of `sd_journal_get_realtime_usec()` and `sd_journal_get_monotonic_usec()` are equivalent. How about to introduce `sd_journal_get_dual_timestamp()`?

**Comment by poettering** [comment]
hmm, i don't grok the rationale? so far we showed the most accurate time we have but use the order of reception as ordering. Why isn't that OK? Can you elaborate?

**Comment by poettering** [comment]
(I mean, people might be confused about that initially, but asynchronous systems are not that hard to grok, so maybe we should just document that?)

we could also add some switch to journalctl to show by reception time, never sender time, but I'd really just make that an option only, that people have to opt in?

or what am I missing?

**Comment by DaanDeMeyer** [comment]
Hmm, I agree this probably isn't ideal, since logs will still be shown out of order, let's close for now
